src/System.Runtime.Numerics/src/System/Numerics/BigInteger.cs

```diff
@@ -552,17 +552,8 @@ public BigInteger(ulong value)
         }
 
         public BigInteger(Single value)
+            : this((Double)value)
         {
-            if (Single.IsInfinity(value))
-                throw new OverflowException(SR.Overflow_BigIntInfinity);
-            if (Single.IsNaN(value))
-                throw new OverflowException(SR.Overflow_NotANumber);
```

Comment:
Note that in these cases:
float.IsInfinity(float.MaxValue * 2.0f) // true
double.IsInfinity(float.MaxValue * 2.0f) // false
The argument is evaluated as double (because that's the type used by default to evaluate floating point expressions). The first case is converted back to float to check for Infinity, but in the second case, the argument is double and is not converted.